File: app/src/types/catalogo.ts
export interface CatalogoFiltrosAplicados {
  regiao: string | null
  uf: string | null
}

export interface CatalogoFiltros {
  filtros_aplicados: CatalogoFiltrosAplicados
  regioes: string[]
  ufs: string[]
  anos: number[]
  quadrantes: string[]
  faixas_confianca: string[]
  faixas_robustez: string[]
  cenarios: string[]
}

export interface CatalogoFiltrosQuery {
  regiao?: string
  uf?: string
}

File: backend/app/schemas/catalogo.py
from pydantic import BaseModel


class FiltrosAplicados(BaseModel):
    regiao: str | None = None
    uf: str | None = None


class CatalogoFiltrosResponse(BaseModel):
    filtros_aplicados: FiltrosAplicados
    regioes: list[str]
    ufs: list[str]
    anos: list[int]
    quadrantes: list[str]
    faixas_confianca: list[str]
    faixas_robustez: list[str]
    cenarios: list[str]


class ReadinessResponse(BaseModel):
    status: str
    database: str
